cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,77 +1,573 @@
+The log messages emitted by your application code can be retrieved by:
+
+- Using the :ref:`restful-api`: the :ref:`Logging HTTP RESTful API
+  <http-restful-api-logging>` details the available contexts that can be called to
+  retrieve different messages.
+- Log messages of a program can be viewed in the :ref:`CDAP UI <cdap-ui>`. In the
+  *Overview* page, select the application and program that you are interested in, and click
+  its *Logs* icon (circled in red here):
+
+  .. figure:: /_images/logging/logging-cdap-ui-purchase.png
+      :figwidth: 100%
+      :width: 800px
+      :align: center
+      :class: bordered-image
+
+      **CDAP UI:** *Log* icon, enabled if there are logs available for viewing 
+
+  The logs will be displayed in the log viewer:
+
+  .. figure:: /_images/logging/logging-cdap-ui-purchase-flow-log.png
+      :figwidth: 100%
+      :width: 800px
+      :align: center
+      :class: bordered-image
+
+      **CDAP UI:** *Log Viewer*, showing *PurchaseFlow* log events, *INFO* level
+
+Program Log File Locations
+--------------------------
+Program logs are stored in locations specified by properties in the
+:ref:`cdap-site.xml <appendix-cdap-default-logging>` file depending on the mode 
+of CDAP (Standalone or Distributed):
+
+- For **Standalone CDAP:** the property ``log.collection.root`` (default
+  ``${local.data.dir}/logs``) is the root location for collecting program logs when in
+  Standalone CDAP.
+
+- For **Distributed CDAP:** the property ``log.base.dir`` (default ``/logs/avro``) is the
+  base directory in HDFS for program logs
+
+.. _configuring-user-application-program-logging:
+
+Configuring Program Logs and Log Levels
+---------------------------------------
+The logging of an application's programs are configured by the ``logback-container.xml``
+file, packaged with the CDAP distribution. This "logback" does log rotation once a day at
+midnight and expires logs older than 14 days. Changes can be made to
+``logback-container.xml``; afterwards, applications or programs needs to be restarted
+for the modified logback file to take effect. Changing the ``logback-container.xml`` will
+only affect programs that are started after the change; existing running programs will not
+be affected.
+
+- For **Distributed CDAP:** the ``logback-container.xml`` file is located in ``/etc/cdap/conf``.
+- For **Standalone CDAP:** as containers are replaced with Java threads, a single
+  ``logback.xml`` file, located in  ``<cdap-sdk-home>/conf``, configures both "container"
+  and CDAP system service logging.
+  
+You can also use a custom "logback" file with your application, as described in the
+Developers' Manual section :ref:`application-logback`.
+
+Changing Program Log Levels
+---------------------------
+The CDAP :ref:`Logging HTTP RESTful API
+<http-restful-api-logging-changing-program-log-levels>` can be used to set the log levels
+of a program, either before a particular run or while it is running. Once changed, they
```

Comment:
To change log levels of a running program, the program type has to be one of service, worker, flow. Other program types are currently not supported. Their log levels can only be changed using preferences before the program starts.